smart-contracts-tutorial/terms-and-invitation.md

```diff
@@ -0,0 +1,89 @@
+# Terms and invitation
+
+This tutorial builds on a [previous tutorial](./first-contract.md). You may want to get familiar with it before continuing reading this page
+
+## Configuring the number to be guessed
+
+We want to configure the contract so that it can be set up for the player to guess an arbitary number
+
+That's when we would be using the `terms`:
```

Comment:
I don't understand your comment

From [what i can read](https://github.com/Agoric/ERTP/blob/98e88d5eba50dc15972b7c174fecdb3c71c16fcc/core/contractHost.chainmail#L56), the argument to `spawn` is `terms`

> I think the value to guess should be an argument to spawn()

This is what i show in the code:

`const { playerInvite } = await E(guessInstallation).spawn({toGuess: 94});`

> Except the terms are visible to all participants

I don't see by which means. From what i read of the ERTP doc (chainmails), there is no method that allows a participant (someone holding a seat) to access the terms directly as they were passed to `.spawn`

... that said, i understand that in more real-world example (with complete information about the contract and not a guessing game which interest is in one participant not having access to all information), every participant should have access to all terms beforehand

Hmm... i wonder whether this is an ERTP bug (the fact that all participants may not have access to all terms). Should a method be added so that a person with a valid invite can also recover the terms as they were when `spawn` was called?

Reply:
You're absolutely right, we didn't document this at all. 

I spent some time adding checkAmount() to the Installation, so people wouldn't have to manually unpack that amount. ContractHost builds the amount associated with an invite with `installation, terms, seatIdentity, seatDesc` in it, so the recipient of an invitation can validate the terms of the contract they've been invited to. This is the way our framework provides assurance to the recipient to an invitation to participate in an auction or a covered call that the contract is for a particular item or amount, so they'd be willing to send it their money.

I will add something to the chainmail description. You looked in the right place, and that info should have been there.

Reply:
ok cool, thanks for the fix in the doc. It makes more sense to me now

The consequence is that terms are public, so maybe the guessing game i proposed does not make sense. Not configured by the terms at least.
I'm going to think about it a bit more